### PR TITLE
Release 2.0.1

### PR DESCRIPTION
**Pull request by zzeekk**
- rename columns for join conditions to avoid ambiguous column errors

### CODE REVIEW

**Comment by kaelte** [comment]
2.11: *** 93 TESTS FAILED ***
2.12: *** 92 TESTS FAILED ***

On my local linux laptop I get many errors like this:
```
LinearDoubleQueryUtilTest:
  org.apache.spark.SparkException: Exception thrown in awaitResult:
  at org.apache.spark.util.ThreadUtils$.awaitResult(ThreadUtils.scala:226)
  at org.apache.spark.sql.execution.exchange.BroadcastExchangeExec.doExecuteBroadcast(BroadcastExchangeExec.scala:146)
  at org.apache.spark.sql.execution.InputAdapter.doExecuteBroadcast(WholeStageCodegenExec.scala:387)
  at org.apache.spark.sql.execution.SparkPlan$$anonfun$executeBroadcast$1.apply(SparkPlan.scala:144)
  at org.apache.spark.sql.execution.SparkPlan$$anonfun$executeBroadcast$1.apply(SparkPlan.scala:140)
  at org.apache.spark.sql.execution.SparkPlan$$anonfun$executeQuery$1.apply(SparkPlan.scala:155)
  at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
  at org.apache.spark.sql.execution.SparkPlan.executeQuery(SparkPlan.scala:152)
  at org.apache.spark.sql.execution.SparkPlan.executeBroadcast(SparkPlan.scala:140)
  at org.apache.spark.sql.execution.joins.BroadcastHashJoinExec.prepareBroadcast(BroadcastHashJoinExec.scala:117)
  ...
  Cause: java.lang.IllegalArgumentException: Unsupported class file major version 55
  at org.apache.xbean.asm6.ClassReader.<init>(ClassReader.java:166)
  at org.apache.xbean.asm6.ClassReader.<init>(ClassReader.java:148)
  at org.apache.xbean.asm6.ClassReader.<init>(ClassReader.java:136)
  at org.apache.xbean.asm6.ClassReader.<init>(ClassReader.java:237)
  at org.apache.spark.util.ClosureCleaner$.getClassReader(ClosureCleaner.scala:50)
  at org.apache.spark.util.FieldAccessFinder$$anon$4$$anonfun$visitMethodInsn$7.apply(ClosureCleaner.scala:845)
  at org.apache.spark.util.FieldAccessFinder$$anon$4$$anonfun$visitMethodInsn$7.apply(ClosureCleaner.scala:828)
  at scala.collection.TraversableLike$WithFilter$$anonfun$foreach$1.apply(TraversableLike.scala:733)
  at scala.collection.mutable.HashMap$$anon$1$$anonfun$foreach$2.apply(HashMap.scala:134)
  at scala.collection.mutable.HashMap$$anon$1$$anonfun$foreach$2.apply(HashMap.scala:134)
  ...
```

**Comment by zzeekk** [comment]
>   Cause: java.lang.IllegalArgumentException: Unsupported class file major version 55

Hi @kaelte, this happens if you have an old java version installed. Can you run the tests with "mvn test" and type "java --version" on the same command line?
Also this PR updates Spark to 3.2.1, and "Java 8 prior to version 8u201 support is deprecated as of Spark 3.2.0", see also https://spark.apache.org/docs/3.2.1/#downloading

**Comment by kaelte** [comment]
> > Cause: java.lang.IllegalArgumentException: Unsupported class file major version 55
> 
> Hi @kaelte, this happens if you have an old java version installed. Can you run the tests with "mvn test" and type "java --version" on the same command line? Also this PR updates Spark to 3.2.1, and "Java 8 prior to version 8u201 support is deprecated as of Spark 3.2.0", see also https://spark.apache.org/docs/3.2.1/#downloading

Thanks @zzeekk  for your advice.

# SBB machine
On my SBB machine, `mvn -U clean install` builds and tests branch develop successfully
```
λ java -version
openjdk version "1.8.0_151-1-ojdkbuild"
OpenJDK Runtime Environment (build 1.8.0_151-1-ojdkbuild-b12)
OpenJDK 64-Bit Server VM (build 25.151-b12, mixed mode)
```

# local Linux laptop
Is the same Java used for compiling and running tests?
## Command Line
Build works but tests do not run using
```
$java --version
openjdk 11.0.16 2022-07-19
OpenJDK Runtime Environment (build 11.0.16+8-post-Ubuntu-0ubuntu120.04)
OpenJDK 64-Bit Server VM (build 11.0.16+8-post-Ubuntu-0ubuntu120.04, mixed mode, sharing)
```

## IntelliJ
Can build but tests do not run using
```
/usr/lib/jvm/java-1.8.0-openjdk-amd64/bin/java -version
openjdk version "1.8.0_342"
OpenJDK Runtime Environment (build 1.8.0_342-8u342-b07-0ubuntu1~20.04-b07)
OpenJDK 64-Bit Server VM (build 25.342-b07, mixed mode)
```

File: src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala [hunk 1]
```diff
@@ -40,7 +40,7 @@ trait TestUtils extends Logging {
   }
 
   def schemaEqual(df1: DataFrame, df2: DataFrame): Boolean = {
-    df1.schema.toDDL == df2.schema.toDDL // ignore nullability in comparision
+    df1.schema.sql == df2.schema.sql // ignore nullability in comparison
```
Comment (kaelte) [inline src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala:42]:
```suggestion
  def schemaEqual(df1: DataFrame, df2: DataFrame): Boolean = {
    def makeFieldNullable(sf: StructField): StructField = StructField(sf.name, sf.dataType, true, sf.metadata)
    df1.schema.map(makeFieldNullable) == df2.schema.map(makeFieldNullable) // ignore nullability in comparison
  }
```
I would rather prefer a typed comparison even if the code is longer.

Comment (kaelte) [reply src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala:42]: Also optimise imports please, i.e. remove
```
import java.sql.Timestamp
import org.apache.spark.sql.types.{DataType, DataTypes, StructType}
```

**Review by kaelte — CHANGES_REQUESTED** [review]
some minor improvements possible

Optimize imports and reformat code using IntelliJ: <Ctrl>-O , <Ctrl>-L
which can be done automatically while committing.

**Comment by kaelte** [comment]
On my local Linux laptop I manage to build and run the tests, but only for Scala 2.12:
`mvn -P scala-2.11 -U clean install` fails
`mvn -P scala-2.12 -U clean install` succeeds.

Same behaviour in IntelliJ.

**Comment by zzeekk** [comment]
> On my local Linux laptop I manage to build and run the tests, but only for Scala 2.12: `mvn -P scala-2.11 -U clean install` fails `mvn -P scala-2.12 -U clean install` succeeds.

I was able to reproduce this locally. The problem is that Scala 2.11 is on Spark 2.4.x, which does not yet support Java 11. So you need a Java 8 SDK installed to compile for Scala 2.11 Actionally thats also what our Github Action uses.
Luckily Java 8 still works for Scala 2.12 and Spark 3.x :-)

I'll add a note about that in the Readme.

Imports are optimized.

File: src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala (hunk 1 above)
Comment (zzeekk) [reply src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala:42]: @kaelte thanks for the suggestion, but your `def makeFieldNullable` just compares the top level attributes. There might be nested structs where we need to ignore nullability and metadata as well. I would stay with the short version `.sql` which can handle these cases, as it is only for testing.

Comment (zzeekk) [reply src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala:42]: Imports optimized.

Comment (kaelte) [reply src/test/scala/ch.zzeekk.spark.temporalquery/TestUtils.scala:42]: > `makeFieldNullable`

Yes, you are right. We can stick with `.sql` or use the same method as in SDL:
https://github.com/smart-data-lake/smart-data-lake/blob/develop-spark3/sdl-core/src/main/scala/io/smartdatalake/util/misc/SchemaUtil.scala#L74

**Review by kaelte — APPROVED** [review]

**Pull request merged**